GitCommands/CommitDataManager.cs

```diff
@@ -157,79 +136,58 @@ public CommitData CreateFromFormatedData(string data)
-
+            var parentGuids = lines[2].Split(' ');
             var author = module.ReEncodeStringFromLossless(lines[3]);
             var authorDate = DateTimeUtils.ParseUnixTime(lines[4]);
-
             var committer = module.ReEncodeStringFromLossless(lines[5]);
             var commitDate = DateTimeUtils.ParseUnixTime(lines[6]);
-
-            string commitEncoding = lines[7];
-
-            const int startIndex = 8;
-            string message = ProccessDiffNotes(startIndex, lines);
+            var commitEncoding = lines[7];
+            var message = ProcessDiffNotes(startIndex: 8, lines);
 
             // commit message is not reencoded by git when format is given
             var body = module.ReEncodeCommitMessage(message, commitEncoding);
 
-            var commitInformation = new CommitData(guid, treeGuid, parentGuids, author, authorDate,
-                committer, commitDate, body);
-
-            return commitInformation;
+            return new CommitData(guid, treeGuid, parentGuids, author, authorDate, committer, commitDate, body);
         }
 
-        /// <summary>
-        /// Creates a CommitData object from formated commit info data from git.  The string passed in should be
-        /// exact output of a log or show command using --format=LogFormat.
-        /// </summary>
-        /// <param name="data">Formated commit data from git.</param>
-        public void UpdateBodyInCommitData(CommitData commitData, string data)
-        {
-            if (data == null)
-            {
-                throw new ArgumentNullException(nameof(data));
-            }
-
-            var module = GetModule();
-
-            var lines = data.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);
-
-            var guid = lines[0];
-
-            string commitEncoding = lines[1];
-
-            const int startIndex = 2;
-            string message = ProccessDiffNotes(startIndex, lines);
-
-            // commit message is not reencoded by git when format is given
-            Debug.Assert(commitData.Guid == guid, "commitData.Guid == guid");
-            commitData.Body = module.ReEncodeCommitMessage(message, commitEncoding);
-        }
-
-        /// <summary>
-        /// Creates a CommitData object from Git revision.
-        /// </summary>
-        /// <param name="revision">Git commit.</param>
-        /// <returns>CommitData object populated with parsed info from git string.</returns>
+        /// <inheritdoc />
         public CommitData CreateFromRevision(GitRevision revision)
         {
             if (revision == null)
             {
                 throw new ArgumentNullException(nameof(revision));
             }
 
-            CommitData data = new CommitData(revision.Guid, revision.TreeGuid, revision.ParentGuids.ToList().AsReadOnly(),
+            return new CommitData(revision.Guid, revision.TreeGuid, revision.ParentGuids.ToList().AsReadOnly(),
```

Comment:
While you are optimizing.. string.Format could be replaced with string.Concat. string.Format has a rather high overhead of parsing a template. In this particular case strings can simply be concatenated.

Reply:
Did some BenchmarkDotNet measurements and the differences aren't as high as I thought. string.Concat is faster but by only ~20%. 254ms vs 308ms in my specific test case.

Reply:
http://www.tomdupont.net/2017/03/c-string-interpolation-performance.html

Reply:
Differences as percentages for this kind of thing seem more significant than they are in reality. A handful of milliseconds difference for a million operations isn't worth worrying about IMO. For that many operations, you're churning so much memory that you have other problems. Go for whatever is easiest to read. In this case, I would convert the lines just beneath your comment to string interpolation. There are many places in the current code that I'd use interpolation just for the clarity it brings, but that's another PR for another day.

Reply:
I agree with your position for high level code / code which is executed relatively rarely. But for code which is executed 1000s of times every single optimization counts because these small seemingly unimportant details add up.
In this specific case it seems I've made a mistake assuming this would be executed on all parsed history revisions.

Reply:
> Go for whatever is easiest to read

Amen. I choose readability and maintainability over optimisation unless it is proven to be a bottle neck